### PR TITLE
[record-minmax-thread-safety-test] Introduce thread safety test

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR adds thread safety test for parallel record-minmax.

for issue https://github.com/Samsung/ONE/issues/10133
from draft: https://github.com/Samsung/ONE/pull/10132

Note: this pr should be rebased after https://github.com/Samsung/ONE/pull/10300 will be merged

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Comment by BalyshevArtem** [comment]
Ready to review

File: compiler/record-minmax-thread-safety-test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
+if(NOT ENABLE_TEST)
+    return()
+endif(NOT ENABLE_TEST)
+
```
Comment (jinevening) [inline compiler/record-minmax-thread-safety-test/CMakeLists.txt:4]: Please enable this test only in 64bit arch.

See https://github.com/Samsung/ONE/pull/10317 for details.

Comment (BalyshevArtem) [reply compiler/record-minmax-thread-safety-test/CMakeLists.txt:4]: Done, thank you!

**Review by jinevening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**